doc/source/data_modules.rst

```diff
@@ -3,164 +3,157 @@
+
+
+Image Definition Modules
+------------------------
+
+Any directory under :file:`data` that is not `scripts` or `overlayfiles`
+is considered an image definition data module and may be included in the
+main image definition using the `_include` statement.
+
+Inheritance rules apply similarly to the image definition tree, but
+additionally, `keg` supports sub-versions of data modules. This can be used for
+instance to create slightly different versions of modules for use with
+different image versions while still sharing most of the image definition
+between those versions.
+
+For this purpose, the `keg` supports the `include-paths` directive in the image
+definition. Include paths are paths that get appended to any source path and
+those get scanned for input files as well. See the following image definition
+as an example:
 
 .. code:: yaml
 
   include-paths:
     leap15/1
     leap15/2
-  profiles:
-    common:
-      include:
-        base/common
+  image:
+    preferences:
+      - _include:
+          - base/common
+    packages:
+      - _include:
+          - base/common
+    config:
+      - _include:
+          - base/common
 
 
 This tells `keg`, when adding data from directory :file:`data/base/common` to
 the image data dictionary, to also look into sub directories :file:`leap15/2`,
 :file:`leap15/1`, and :file:`leap15` (through inheritance). This would lead to
 the following directories being scanned::
 
+  data
   data/common
   data/common/base
   data/common/base/leap15
   data/common/base/leap15/1
   data/common/base/leap15/2
 
 This allows for example to put generic configuration bits in
-:file:`common/base`, Leap 15 specific configuration in
-:file:`common/base/leap15`, and adjust the configuration for minor versions, if
-necessary.
+:file:`data/common/base`, Leap 15 specific configuration in
+:file:`data/common/base/leap15`, and adjust the configuration for minor
+versions, if necessary.
 
-When building the dictionary, `keg` will parse all input files referenced
-in the `profiles` section and merge them into the main dictionary. The
-following section describes the structure used in the data section.
+When merging the included dictionaries into the main dictionary, `keg` only
+copies the dictionary under the top level key that matches the key under
+which the `_include` statement is. That mean, assuming the YAML files collected
+from the above trees resulted in the following dictionary:
 
+.. code:: yaml
 
-Data Module Dictionary Structure
---------------------------------
-
-This section describes the input parameters used by the data modules.
-
-.. note::
+  preferences:
+    locale: en_US
+    timezone: UTC
+    type:
+      _attributes:
+        firmware: efi
+        image: vmx
+  packages:
+    _namespace_base_packages:
+      package:
+        - bash
+        - glibc
+        - kernel-default
+  config:
+    _namespace_base_services:
+      services:
+        - sshd
```

Comment:
This doesn't match. The text states "...in the following dictionary" but hen we show a YAML representation. WOrding and what we show needs to match.

Reply:
Replaced with "data structure".